distributed/client.py

```diff
@@ -3826,108 +3826,101 @@ def scheduler_info(self, **kwargs):
-            if not filename.endswith(suffix):
-                filename += suffix
+        with opener(filename, mode) as f:
+            write(state, f)
 
-            with open(filename, "w") as fd:
-                yaml.dump(state, fd)
-        else:
-            raise ValueError(
-                f"Unsupported format {format}. Possible values are `msgpack` or `yaml`"
-            )
+    async def _dump_cluster_state_remote(
+        self,
+        filename: str,
+        exclude: Collection[str],
+        format: Literal["msgpack", "yaml"],
+        storage_options: dict | None = None,
+    ) -> None:
+        "Tell the scheduler to dump cluster state to a URL"
+        await self.scheduler.dump_cluster_state_to_url(
+            url=filename,
+            exclude=exclude,
+            format=format,
+            storage_options=storage_options,
+        )
 
     def dump_cluster_state(
         self,
         filename: str = "dask-cluster-dump",
         exclude: Collection[str] = ("run_spec",),
         format: Literal["msgpack", "yaml"] = "msgpack",
+        **kwargs,
     ):
-        """Extract a dump of the entire cluster state and persist to disk.
+        """Extract a dump of the entire cluster state and persist to disk or a URL.
         This is intended for debugging purposes only.
 
-        Warning: Memory usage on client side can be large.
+        Warning: Memory usage on the scheduler (and client, if writing the dump locally)
+        can be large. On a large or long-running cluster, this can take several minutes.
+        The scheduler may be unresponsive while the dump is processed.
 
         Results will be stored in a dict::
 
             {
-                "scheduler_info": {...},
-                "worker_info": {
-                    worker_addr: {...},  # worker attributes
+                "scheduler": {...},  # scheduler state
+                "workers": {
+                    worker_addr: {...},  # worker state
                     ...
                 }
+                "versions": {
+                    "scheduler": {...},
+                    "workers": {
+                        worker_addr: {...},
+                        ...
+                    }
+                }
             }
 
         Parameters
         ----------
         filename:
-            The output filename. The appropriate file suffix (`.msgpack.gz` or
-            `.yaml`) will be appended automatically.
+            The output filename. The appropriate file suffix (``.msgpack.gz`` or
+            ``.yaml``) will be appended automatically.
+
+            If the filename is a URL supported by :func:`fsspec.open` (like
+            ``s3://my-bucket/cluster-dump``, or any URL containing ``://``), the
+            dump will be written directly to that URL from the scheduler.
+            Any additional keyword arguments will be passed to :func:`fsspec.open`.
+            Note that this requires having ``fsspec`` (and any implementation like
```

Comment:
As below, `fsspec` should be unconditionally available, so this note should just be referring to `s3fs` etc.